### ISSUE TITLE
Disable colorization by default if output is not a text terminal

### ISSUE THREAD

**Issue by specious** [fazibear/colorize#53]
By convention, I am used to colorized applications suppressing colorization when sending the output to a pipe, unless I explicitly force colorization.

For example, in [colors.js](https://github.com/Marak/colors.js/blob/9f3ace44700b8e705cb15be4767845c311b3ae11/lib/system/supports-colors.js#L40):

```js
if (process.stdout && !process.stdout.isTTY) {
  return false;
}
```

Writing an app using `colorize`, I was surprised that colors were not suppressed when doing `app > file.txt` or `app | cat`.  The color codes can cause problems in later processing stages.

I then [turned colorization off](https://github.com/specious/meetup-cli/blob/4020e087c717ee2badfc235b14948f5756ef9f7f/lib/meetup-cli.rb#L19) for non-TTY output with:

```ruby
String.disable_colorization(true) unless STDOUT.isatty
```

I'd like to propose that be the default behavior, with a way to force colors by choice in all cases.

**Comment by fazibear** [comment]
It worked like you described, but it was changed few versions ago because of PR or issue.

**Comment by specious** [comment]
I see [in the CHANGELOG](https://github.com/fazibear/colorize/blob/468c2f7832239dbac30db51c889c4bd3378f1766/CHANGELOG#L42):

```
== 0.6.0 / 2013-09-25
  ...
  * STDOUT.isatty condition removed
```

What were the discussions relevant to this decision?

**Comment by fazibear** [comment]
As you can see it was 3 years ago :) Can't find it, it was a PR or issue.

**Comment by specious** [comment]
Here's the problem.  This looks nice:
![meetup-cli-color-output](https://cloud.githubusercontent.com/assets/451178/19872611/15040366-9f91-11e6-921f-1069cc5543c4.png)
On the other hand, this is an unpleasant surprise:
![meetup-cli-color-codes-problem](https://cloud.githubusercontent.com/assets/451178/19872624/2528824e-9f91-11e6-9ae2-d6b9e15a6a79.png)
In my experience, the default behavior is usually implemented to be the opposite of what it is here.

**Comment by ccoenen** [comment]
@specious in case this bugs you on an ongoing basis: less knows how to interpret the codes if you start it with `less -r`. Many other cli tools have a similar option.

If this comes back, it would sure be nice to be able to override this again (kind of like `git diff --color=always`) so one can choose to pipe to a log file and _still_ get color codes.

**Comment by jefflunt** [comment]
Makes sense to me to go back to at least consider going back to the previous functionality. I don't think I've ever dumped `colorize` strings to a log, but that's where I can think that this would be most painful, especially if said log was being read back from a log aggregator that didn't know what do with all the escape sequences.

If the PRs I submitted today look interesting to @fazibear and have a chance at getting merged in the next few months, then I might consider digging through the PR / issue history to look at when this was last working the way the issue asks in order to see if I can figure out how to restore the previous capability.

No promises, but I get why this is valuable.

(NOTE: yes, I definitely realize that this issue is now over 5 years old)

**Comment by chocolateboy** [comment]
> I might consider digging through the PR / issue history to look at when this was last working the way the issue asks

FYI, the change was introduced in this PR: https://github.com/fazibear/colorize/pull/2

**Comment by jefflunt** [comment]
@chocolateboy - thanks for finding that! I see [this PR](https://github.com/skycocker/chromebrew/issues/4554) from the `chromebrew` project also mentioned the change to this gem, and that they re-introduced the TTY-sensitivity on their own.

What's your take on #2 and the reason for removing this originally?

I created [a branch with this change](https://github.com/jefflunt/colorize/blob/consider-tty/lib/colorize/instance_methods.rb#L20-L22) on my fork to test.

**Comment by chocolateboy** [comment]
I think #2 should be opt-in (like [`--color=always`](https://github.com/BurntSushi/ripgrep/blob/64ac2ebe0f2fe1c8967e7ec550bc32466bf40a07/FAQ.md#how-do-i-configure-ripgreps-colors) in ripgrep). It's a reasonable feature ("Need an option to colorize output when writing to a file"), but the implementation has inadvertently introduced a bug ("No way to disable color when writing to a file").
